packages/graalvm/src/main/kotlin/elide/runtime/gvm/internals/node/fs/NodeFilesystem.kt

```diff
@@ -77,8 +77,12 @@ import elide.vm.annotations.Polyglot
   @Singleton fun providePromises(): FilesystemPromiseAPI = promises
 
   override fun install(bindings: MutableIntrinsicBindings) {
-    bindings[NodeFilesystem.SYMBOL_STD.asJsSymbol()] = std
-    bindings[NodeFilesystem.SYMBOL_PROMISES.asJsSymbol()] = promises
+    bindings[NodeFilesystem.SYMBOL_STD.asJsSymbol()] = ProxyExecutable { provideStd() }
+    bindings[NodeFilesystem.SYMBOL_PROMISES.asJsSymbol()] = ProxyExecutable { providePromises() }
```

Comment:
FS bindings are now deferred until the module is initialized, because the VFS instance is not yet available when the binding is first installed.

Reply:
I like it